components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -3086,8 +3078,8 @@ public static String getIdTokenIssuer(String tenantDomain) throws IdentityOAuth2
         FederatedAuthenticatorConfig oidcAuthenticatorConfig =
                 IdentityApplicationManagementUtil.getFederatedAuthenticator(fedAuthnConfigs,
                         IdentityApplicationConstants.Authenticator.OIDC.NAME);
-        return IdentityApplicationManagementUtil.getProperty(oidcAuthenticatorConfig.getProperties(),
-                IDP_ENTITY_ID).getValue();
+        return IdentityUtil.resolveURL(IdentityApplicationManagementUtil
```

Comment:
I think this is not needed. What we do here is to return the value set at the resident IDP. It can be a random URI value. Trying to tenant qualify that is incorrect.

Reply:
For example, I set a value like "https://myissuer.xyz.com" at the resident IDP as the issuer value. When we use the IdentityUtil.resolveURL() method what would be the output?

Reply:
fixed with https://github.com/wso2/carbon-identity-framework/pull/2836/commits/f8b4a59dc81f0b5f1d29aa06e3109d6b80721781